### ISSUE TITLE
Fake accents

### ISSUE THREAD

**Issue by Enivex** [typst/typst#422]
Currently https://typst.app/docs/reference/math/accent/ will only display correctly if supplied with one of the actually supported accents.

In LaTeX I use https://ctan.org/pkg/accents?lang=en, which can be used to typeset more or less anything as an accent. It would be useful if Typst's accent function would revert to similar behavior when supplied with something other than the few supported symbols.

**Comment by user202729** [comment]
You can already stack the boxes manually, but I don't know how to do that reliably while correctly handle the spacing.

**Comment by alerque** [comment]
Don't go down this road. The approach LaTeX used is a bandaid on a hack. The correct technology exists today: OpenType fonts can specify the placement of accents on characters in a robust and correct way. If you want the support, use a font that supports it. The typesetter should not be responsible for faking it. The low level tools can be provided to fake something up, but it should not be provided as a general purpose tool to handle accented characters.

**Comment by Enivex** [comment]
> Don't go down this road. The approach LaTeX used is a bandaid on a hack. The correct technology exists today: OpenType fonts can specify the placement of accents on characters in a robust and correct way. If you want the support, use a font that supports it. The typesetter should not be responsible for faking it. The low level tools can be provided to fake something up, but it should not be provided as a general purpose tool to handle accented characters.

I think you misunderstand. The point is that even symbols that aren't normally used as accents can be used as them. Obviously regular accents wouldn't be faked.

Here's an example:

![image](https://user-images.githubusercontent.com/2191754/228509013-b354fae8-185e-4b28-a84e-f38a0c033cd9.png)

**Comment by Enivex** [comment]
> You can already stack the boxes manually, but I don't know how to do that reliably while correctly handle the spacing.

Yes, that's the hard part. I've tried playing around with it, but since I lack the understanding of Typst's intracicies when it comes to spacing and positioning I couldn't make it work properly.
